phalcon/validation/validator/numericality.zep

```diff
@@ -71,7 +71,7 @@ class Numericality extends Validator
 
 		let value = validation->getValue(field);
 
-		if !preg_match("/^-?\d+\.?\d*$/", value) {
+		if !preg_match("/^-?\d+[\.,]?\d*$/", value) || !is_numeric(value) {
```

Comment:
In general looks good but I fear for already working applications that expects that `99,99` is not numeric. Also as I know the `is_numeric` function rely on locale setup. Finally, introducing `is_numeric` will also "say" that `1.45e99` is number (previously not).

Reply:
It will test first is `99.99` or `99,99`, so `is_numeric` used only for checks if `99.` or `99,` not used because it's not a right numeric format. 

>  I fear for already working applications that expects that 99,99 is not numeric.

It exactly a problem, it's a bug. On some locales it must be a proper number format.
Becides `is_numeric` uses locale format, so for `en_US.utf8` number `99,99` is still not valid, but it will be valid for `ru_RU.utf8` which now is not working

Reply:
It test both regexp and `is_numeric`, so in case if one of them returns false - it false,
so it does not except `1.45e99` and will except only valid numbers for current locale

Reply:
I wonder if we should just switch to internally use PHP's `is_numeric(value)` only. That way we can avoid any bias that we've implemented. Instead we should be more of a wrapper around existing and well thought out methods that exist in the core.

Thus, I'm actually in favor of just replacing with: `if !is_numeric(value) {`

Reply:
It appears that `is_numeric()` covers everything in the regular expression and more. So yeah .. we should just straight proxy ... I think that's the best solution.